deploy_vpipe.sh

```diff
@@ -0,0 +1,229 @@
+}
+
+message() {
+	printf '\e[37;1m%s\t%s\e[0m\n' "$1" "$2"
+}
+
+status() {
+	printf '\e[36;1m%s\e[0m\n'	"$1"
+}
+
+check_directory() {
+	if [[ -d "$1" ]]; then
+		if (( FORCE )); then
+			rm -rf "$1"
+		else
+			fail "${2:-Directory} ${1} already exists" 'check and use -f if needed'
+		fi
+	fi
+}
+
+usage() {
+echo "usage: $0 [options]
+options:
+-f           force overwriting directories
+-p PREFIX    prefix directory under which to install V-pipe
+             [default: current directory]
+-b BRANCH    install specified branch of V-pipe${BRANCH:+
+             [default: $BRANCH]}
+-r RELEASE   install specified release package of V-pipe${RELEASE:+
+             [default: $RELEASE]}
+-w WORKDIR   create and populate working directory
+-m           only minimal working directory
+-h           print this help message and exit"
+}
+
+
+# parameters
+while getopts ":fp:b:r:w:mh" opt; do
+	case ${opt} in
+		f)
+			FORCE=1
+		;;
+		p)
+			PREFIX="${OPTARG}"
+		;;
+		b)
+			BRANCH="${OPTARG}"
+		;;
+		r)
+			RELEASE="${OPTARG}"
+		;;
+		w)
+			WORKDIR="${OPTARG}"
+		;;
+		m)
+			MINIMAL='-m'
+		;;
+		h)
+			usage
+			exit 0
+		;;
+		\?)
+			fail "Invalid option: ${OPTARG}"
+		;;
+		:)
+			fail "Invalid option: ${OPTARG} requires an argument"
+		;;
+	esac
+done
+shift $((OPTIND -1))
+
+[[ -z "${BRANCH}" && -z "${RELEASE}" ]] && fail "Please specify either a branch or a release package"
+
+
+###################
+#                 #
+#   Basic stuff   #
+#                 #
+###################
+
+DOWNLOAD=
+if [[ -x $(which wget) ]]; then # most Linux distros (Gentoo)
+	DOWNLOAD='wget'
+elif [[ -x $(which curl) ]]; then # a few Linux distros (CentOS, Archlinux) and Mac OS X
+	DOWNLOAD='curl -O'
+else
+	fail 'Please install either wget or curl'
+fi;
+
+
+
+##################
+#                #
+#   Miniconda3   #
+#                #
+##################
+
+MINICONDA=
+MINICONDAPATH="${PREFIX}/miniconda3"
+
+title 'installing Miniconda3'
+
+# Check if directory is free
+check_directory "${MINICONDAPATH}" 'Miniconda3 installation path'
+
+# Check OS for OS-Spefic Miniconda3 installer
+if [[ "$OSTYPE" == linux* ]]; then 
+	MINICONDA=Miniconda3-latest-Linux-x86_64.sh 
+elif [[ "$OSTYPE" == darwin* ]]; then
+	MINICONDA=Miniconda3-latest-MacOSX-x86_64.sh
+else
+	fail 'I cannot detect OS. Only Linux and Mac OS X are supported' 'manually override OSTYPE environment variable if needed'
+fi
+
+message 'Using installer:' ${MINICONDA}
+
+# Get and install Miniconda3
+mkdir -p ${PREFIX}
+cd ${PREFIX}
+[[ -f ${MINICONDA} ]] && rm ${MINICONDA}
+${DOWNLOAD} https://repo.anaconda.com/miniconda/${MINICONDA}
+sh ${MINICONDA} -b -p miniconda3
+# -b for batch (no question asked)
+
+
+. miniconda3/bin/activate
+
+# set the channel precedence (lowest to highest)
+conda config --add channels defaults
+conda config --add channels bioconda
+conda config --add channels conda-forge 
+# NOTE conda-forge *HAS TO* be higher than bioconda
```

Comment:
Enforcing the order of channels is critical:
according to bioconda authors, we absolutely need to make use that the conda-forge channel has a higher precedence over bioconda.